### PR TITLE
Node.js example with dependency handling and hot-reload

### PR DESCRIPTION
**Pull request by mfornasa**
This is an implementation of the new file sync feature with Node.js. 
It's more realistic than current one in `hot-reload` dir, as it features dependency handling with `npm` and `package.json`.

### CODE REVIEW

**Comment by r2d4** [comment]
Thanks! I’ll take a look soon

File: integration/examples/nodejs/README.adoc [hunk 1]
```diff
@@ -0,0 +1,16 @@
+=== Example: Node.js with hot-reload
+:icons: font
+
+Simple example based on Node.js demonstating the file synchronization mode.
```
Comment (dgageot) [inline integration/examples/nodejs/README.adoc]: s/demonstating/demonstrating

File: integration/examples/nodejs/backend/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
+FROM node:8.12.0-alpine
+
+WORKDIR /opt/backend
+
+COPY . /opt/backend
```
Comment (dgageot) [inline integration/examples/nodejs/backend/Dockerfile]: Can be `COPY . .`

File: integration/examples/nodejs/backend/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+RUN npm install
+
+CMD ["npm", "run", "dev"]
```
Comment (dgageot) [inline integration/examples/nodejs/backend/Dockerfile]: Can be moved up just after `WORKDIR`

**Comment by mfornasa** [comment]
*@dgageot Do you prefer an additional commit or a rebased one?*

**Comment by dgageot** [comment]
As you see fit

**Comment by mfornasa** [comment]
@dgageot done

File: integration/examples/nodejs/skaffold.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+build:
+  artifacts:
+  - image: gcr.io/k8s-skaffold/node-example
+    context: backend
+    sync:
+      '*.js': .
+deploy:
+  kubectl:
+    manifests:
+    - "**/k8s"
```
Comment (r2d4) [inline integration/examples/nodejs/skaffold.yaml]: I think you have this backwards. You want `k8s/**`

Comment (r2d4) [reply integration/examples/nodejs/skaffold.yaml]: this path is relative to the skaffold.yaml

File: integration/examples/nodejs/backend/.dockerignore [hunk 5]
```diff
@@ -0,0 +1,2 @@
+node_modules
+*.swp
```
Comment (r2d4) [inline integration/examples/nodejs/backend/.dockerignore]: nit: newline

File: integration/examples/nodejs/skaffold.yaml (hunk 4 above)
Comment (mfornasa) [reply integration/examples/nodejs/skaffold.yaml]: You're right! This is a copy-paste from https://github.com/GoogleContainerTools/skaffold/blob/master/integration/examples/hot-reload/skaffold.yaml#L16, so you may want to take a look there too.

Comment (mfornasa) [reply integration/examples/nodejs/skaffold.yaml]: I changed this to something more explicit

**Pull request merged**